File: app/src/main/java/com/stone/lfernandosantos/storewars/views/SplashScreenActivity.java
package com.stone.lfernandosantos.storewars.views;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;

import com.stone.lfernandosantos.storewars.R;

public class SplashScreenActivity extends AppCompatActivity implements Runnable {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash_screen);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        Handler handler = new Handler();

        handler.postDelayed(this, 3000);

    }

    @Override
    public void run() {
        startActivity(new Intent(this, HomeActivity.class));
        finish();
    }
}

File: app/src/main/java/com/stone/desafiomobile/view/SplashScreenActivity.kt
package com.stone.desafiomobile.view

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.stone.desafiomobile.R
import kotlinx.coroutines.experimental.CommonPool
import kotlinx.coroutines.experimental.delay
import kotlinx.coroutines.experimental.launch

/**
 * Activity com a splash screen
 */
class SplashScreenActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash_screen)

        /**
         * espera um tempo para ir para a proxima activity
         */
        launch(CommonPool) {
            val intent = Intent(this@SplashScreenActivity, MainActivity::class.java)
            delay(500)
            startActivity(intent)
            finish()
        }
    }
}
